### ISSUE TITLE
SINGULARITY_SYPGPDIR not being honored

### ISSUE THREAD

**Issue by khurtado** [apptainer/singularity#5516]
### Version of Singularity:

What version of Singularity are you using? Run:

```
$ singularity version
3.6.0-1.el7

```

### Expected behavior

When setting the `SINGULARITY_SYPGPDIR` environment variable, pgp related files should be writted there, rather than `~/.singularity/sypgp`. 

### Actual behavior

`~/.singularity/sypgp` is created rather than `$SINGULARITY_SYPGPDIR`.

### Steps to reproduce this behavior
Use a .sif image that writes PGP signatures, then:

```
[khurtado@qa-rtx6k-034 ~]$ rm -rf ~/.singularity
[khurtado@qa-rtx6k-034 ~]$ echo $SINGULARITY_CACHEDIR
/tmp/.singularity_khurtado
[khurtado@qa-rtx6k-034 ~]$ echo $SINGULARITY_SYPGPDIR
/tmp/.singularity_khurtado/sypgp
[khurtado@qa-rtx6k-034 ~]$ mkdir -p /tmp/.singularity_khurtado/sypgp
[khurtado@qa-rtx6k-034 ~]$ singularity shell /afs/crc.nd.edu/x86_64_linux/s/singularity/images/pytorch-1.7.sif
Singularity> exit
exit
[khurtado@qa-rtx6k-034 ~]$ ls -l ~/.singularity
total 2
drwx------ 2 khurtado campus 2048 Aug 20 11:44 sypgp
[khurtado@qa-rtx6k-034 ~]$ ls -l $SINGULARITY_SYPGPDIR
total 0
[khurtado@qa-rtx6k-034 ~]$
```

### What OS/distro are you running

```
$ cat /etc/os-release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.8 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.8"
PRETTY_NAME="Red Hat Enterprise Linux"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.8:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.8
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.8"


```

### How did you install Singularity

Write here how you installed Singularity. Eg. RPM, source.
```
rpm -qa | grep singularity
singularity-3.6.0-1.el7.x86_64
https://github.com/hpcng/singularity/releases/tag/v3.6.0
```

**Comment by khurtado** [comment]
Any comments on this?

**Comment by dtrudg** [comment]
Smells like a bug - we'll take a look at it when we can.

**Comment by khurtado** [comment]
Sounds good. Thanks!

**Comment by dtrudg** [comment]
The keyring is being loaded in `prepare_linux.go` for ECL functionality. This happens in a place where the environment var is not available, so we are defaulting to home dir location. Need to be able to pass down a env var set keyring location into the runtime.

**Comment by dtrudg** [comment]
This should be fixed properly via #5637 as the ECL code (which runs where `SINGULARITY_SYPGPDIR` is not propagated) will use only a system-wide public keyring, and won't trigger the creation/loading of things in `$HOME`.

Will verify when merged.

**Comment by dtrudg** [comment]
Verified #5637 fixes this.
